site/docs/pages/swap/types.mdx

```diff
@@ -4,3 +4,19 @@ description: Glossary of Types in Swap Kit.
 ---
 
 # Types [Glossary of Types in Swap Kit.]
+
+## `SwapAmountInputReact`
+
+```ts
+type SwapAmountInputReact = {
+  label: string; // Descriptive label for the input field
+  amount?: string; // Token amount
+  token?: Token; // Selected token
+  swappableTokens?: Token[]; // Tokens available for swap
+  tokenBalance?: string; // Amount of selected Token user owns
+  onMaxButtonClick: () => void; // Callback function when max button is clicked
+  onAmountChange: (amount: string) => void; // Callback function when the amount changes
+  onTokenSelectorClick: () => void; // Callback function when the token selector is clicked
```

Comment:
Maybe let's change the `onTokenSelectorClick` to `setToken`? The user shouldn't have to know that we are using TokenSelector under the hood.

```suggestion
  setToken: () => void; // Callback function when the token selector is clicked
```

Reply:
yes agreed.

Reply:
The other cb props i am passing in start with on i.e `onAmountChange`, `onMaxButtonClick`. Should I do `onTokenSelect` to keep it consistent or do you think `setToken` is more clear ?

Reply:
update: went with `setToken` and `setAmount`